src/llscan.cc

```diff
@@ -1242,12 +1243,20 @@ void LLScan::ScanMemoryRanges(FindJSObjectsVisitor& v) {
       for (size_t j = 0; j + addr_size <= loaded;) {
         uint64_t value;
 
-        if (addr_size == 4)
+        if (addr_size == 4) {
           value = *reinterpret_cast<uint32_t*>(&block[j]);
-        else if (addr_size == 8)
+          if (process_byte_order != HOST_BYTE_ORDER) {
+            value = __builtin_bswap32(value);
+          }
+        }
+        else if (addr_size == 8) {
           value = *reinterpret_cast<uint64_t*>(&block[j]);
-        else
+          if (process_byte_order != HOST_BYTE_ORDER) {
+            value = __builtin_bswap64(value);
+          }
+        } else {
```

Comment:
Since you're here anyway, maybe rewrite this so it won't run afoul of alignment issues on non-x86.  I mean this:
```c++
uint32_t tmp;
memcpy(&tmp, &block[j], sizeof(tmp));
value = tmp;
// do byte swap if needed
```
Fine to do in a separate commit.